src/coffee/collector.coffee

```diff
@@ -0,0 +1,47 @@
+((window, $) ->
+  selectors = {
+    fileOpenListItem: 'a.file-list-link',
+    fileDownloadListItem: 'a.file-list-control',
+    videoPlayListItem: '.views-sidebar-video .playlist-list-item-link',
+    audioPlayListItem: '.views-sidebar-audio .playlist-list-item-link',
+    galleryListItem: '.gallery-list-item-link'
+  }
+
+  collectEvent = (path, type) ->
+    ($ window).trigger 'analytics-collect', [{path: path, type: type}]
+
+  fileOpenListItemClicked = (e) ->
```

Comment:
For files, there's is `data-relpath` attribute that holds the correct path. The `href` attrib points to a facet tab.